packages/govuk-frontend/src/govuk/components/button/button.test.js

```diff
@@ -55,105 +61,121 @@ describe('/components/button', () => {
+      let $button
+
       beforeEach(async () => {
         await goToComponent(page, 'button')
-        await trackClicks(page)
+        $button = await page.$('button')
+        await trackClicks($button)
       })
 
-      it('does not prevent multiple submissions', async () => {
-        await page.click('button')
-        await page.click('button')
+      it('does not prevent multiple clicks', async () => {
+        await $button.click()
+        await $button.click()
 
-        const clicksCount = await getClicksCount(page)
+        const counts = await getButtonTracking($button)
 
-        expect(clicksCount).toBe(2)
+        expect(counts.click).toBe(2)
+        expect(counts.debounce).toBe(0)
       })
     })
 
     describe('using data-attributes', () => {
+      /** @type {import('puppeteer').ElementHandle<HTMLButtonElement>} */
+      let $button
+
       beforeEach(async () => {
         await goToComponent(page, 'button', {
           exampleName: 'prevent-double-click'
         })
-        await trackClicks(page)
+
+        $button = await page.$('button')
+        await trackClicks($button)
       })
 
-      it('prevents unintentional submissions when in a form', async () => {
-        await page.click('button')
-        await page.click('button')
+      it('prevents unintentional clicks', async () => {
+        await $button.click({ count: 2 })
 
-        const clicksCount = await getClicksCount(page)
+        const counts = await getButtonTracking($button)
 
-        expect(clicksCount).toBe(1)
+        expect(counts.click).toBe(2)
+        expect(counts.debounce).toBe(1)
       })
 
       it('does not prevent intentional multiple clicks', async () => {
-        await page.click('button')
-        await page.click('button', { delay: 1000 })
+        await $button.click({ count: 2, delay: debouncedWaitTime })
 
-        const clicksCount = await getClicksCount(page)
+        const counts = await getButtonTracking($button)
 
-        expect(clicksCount).toBe(2)
+        expect(counts.click).toBe(2)
+        expect(counts.debounce).toBe(0)
       })
 
       it('does not prevent subsequent clicks on different buttons', async () => {
         // Clone button to have two buttons on the page
         await page.evaluate(() => {
-          const $button = document.querySelector('button')
-          const $secondButton = $button.cloneNode(true)
+          const $button1 = document.querySelector('button')
+          const $button2 = $button1.cloneNode(true)
 
-          document.querySelector('form').appendChild($secondButton)
+          $button1.parentNode.appendChild($button2)
         })
 
         await page.click('button:nth-child(1)')
         await page.click('button:nth-child(2)')
 
-        const clicksCount = await getClicksCount(page)
+        const counts = await getButtonTracking($button)
 
-        expect(clicksCount).toBe(2)
+        expect(counts.click).toBe(1)
```

Comment:
I'm a bit confused by this one - through the `beforeEach`, we're tracking `$button1`, so that's all good, but with this test we don't seem to know if `$button2` has actually been clicked?

Reply:
I did wonder about that. The old tests didn't either (when it was named `$secondButton`)

Shall we add tracking to both buttons?

Reply:
Definitely - to satisfy the test description we need to ensure the second button was clicked successfully. 

The previous test looks like it was at least aiming to test that there were two valid clicks after clicking each button once.

Reply:
Thanks @domoscargin, can you take a look now?

Reply:
Nice, looks sound!